### PR TITLE
[WIP] - Revamp rules view ui

### PR DESCRIPTION
**Pull request by njirap**

### CODE REVIEW

**Comment by bhearsum** [comment]
This is a nice start, and you're on the right track. There's a few things that need to be improved before we can get this merged:
* The rules shown for any given selection aren't quite right. For example, When I filter on "Firefox, release-cdntest" I seem to get *all* of the Firefox rules, including ones with channels like "beta".
* Some additional post processing needs to happen as well. Globbing needs to be handled such that "beta*" rules show up for any filters where the channel starts with "beta". Empty values should be treated as "*" as well. For example, a rule where the product is empty but the channel is "release" should show up in any filter where the channel is "release".
* The entries in the drop down need to be pared down a bit, specifically:
** We don't want every intersection of product+channel in the rules table to be in there, only those that have at least one rule. For example, I have an entry in the drop down for "Firefox, b2g-ota-test" - which has no rules that apply to it.
** No need to show channels with globs (*) in them. These rules are only useful in the context of a concrete channel.
* When I first load the rules page, all rules are shown. This is good, but there's no way to get back to that after I filter them by a specific product+channel.
* Sorting the entries in the drop down would be great.
* The layout is a bit funky on my laptop - it looks like there's some wrapping going on: http://people.mozilla.org/~bhearsum/sattap/3e26fafd.png. If both of the drop downs can't fit on one row one of them should be moved lower so that they can line up, instead of this ugly wrapping.

**Comment by njirap** [comment]
- [ ] The rules shown for any given selection aren't quite right. For example, When I filter on "Firefox, release-cdntest" I seem to get all of the Firefox rules, including ones with channels like "beta".

- [ ] Some additional post processing needs to happen as well. Globbing needs to be handled such that "beta" rules show up for any filters where the channel starts with "beta". Empty values should be treated as "" as well. For example, a rule where the product is empty but the channel is "release" should show up in any filter where the channel is "release".

- [ ] The entries in the drop down need to be pared down a bit, specifically: ** We don't want every intersection of product+channel in the rules table to be in there, only those that have at least one rule. For example, I have an entry in the drop down for "Firefox, b2g-ota-test" - which has no rules that apply to it. ** No need to show channels with globs (*) in them. These rules are only useful in the context of a concrete channel.

- [ ] When I first load the rules page, all rules are shown. This is good, but there's no way to get back to that after I filter them by a specific product+channel.

- [ ] Sorting the entries in the drop down would be great.

- [ ] The layout is a bit funky on my laptop - it looks like there's some wrapping going on: http://people.mozilla.org/~bhearsum/sattap/3e26fafd.png. If both of the drop downs can't fit on one row one of them should be moved lower so that they can line up, instead of this ugly wrapping.

**Comment by njirap** [comment]
Hehe ok..it's easier for me like this..

**Comment by njirap** [comment]
I am unable to debug the one failing test that says '$scope.pairExists' is not a function..

File: app/js/controllers/rules_controller.js [hunk 1]
```diff
@@ -36,6 +42,25 @@ function($scope, $routeParams, $location, $timeout, Rules, Search, $modal, $rout
...
+  $scope.$watch('pr_ch_filter', function(value) {
+    $scope.pr_ch_selected = value.split(',');
+  });
+
+  $scope.pr_ch_options = ["All rules"];
+  Rules.getProducts().success(function(response_prs) {
+    Rules.getChannels().success(function(response_chs) {
+      response_prs.product.forEach(function(pr) {
+        response_chs.channel.forEach(function(ch) {
+          if (!ch.contains("-") && !ch.contains("*") && $scope.pairExists(pr, ch)){
```
Comment (bhearsum) [inline app/js/controllers/rules_controller.js]: I think what might be happening is that this new block of code is always executed in this controller, but $scope.pairExists only exists if rule_id isn't set (it's in the else block of the if/else that starts at line 8). Maybe this should be in that else block as well? This filter probably doesn't matter for a single rule view.

**Comment by bhearsum** [comment]
This is looking really great! Just a couple small things to address (in addition to the test) and I can get this merged!
- [ ] The rule selection is better, but needs to treat channels like "release-localtest" as real channels. Right now they're lumped in with "release". To spell this out a bit more, if I have a rule with no channel, one with "release", one with "release-localtest", and one with "release\*" I'd expect the following:
  * Filtering on "release" shows the no channel, "release", and "release\*" rules.
  * Filtering on "release-localtest" shows the no channel, "release-localtest", and "release\*" rules.
- [ ] The layout is improved, but the labels need to be restored to them - otherwise it's not clear what they do without experimantion. If you're having trouble making the layout work, consider putting the sort drop down below the filter to save on horizontal space.

(I think Github ate the "\*" on my comment about "beta\*", which may have caused confusion...sorry about that.)

**Comment by bhearsum** [comment]
This looks good! There might be a couple of follow-up pieces after people start using it, but I see no reason this can't land.

**Pull request merged**